bokeh/models/widgets/widget.py

```diff
@@ -17,3 +17,15 @@ class Widget(LayoutDOM):
     ''' A base class for all interactive widget types.
 
     '''
+
+class NeedsJQuery(object):
+    """ A mixin for widgets requiring JQuery and/or JQuery UI. """
+
+    __javascript__ = [
+        "https://code.jquery.com/jquery-3.2.1.min.js",
+        "https://code.jquery.com/ui/1.12.1/jquery-ui.min.js",
+    ]
+
+    __css__ = [
+        "https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css",
+    ]
```

Comment:
Yah I am not certain about the viability of this approach. I think perhaps we have to either:
* re-implement widgets without jq/jq-ui *at all*
* cleave `bokeh-widgets` into two pieces `bokeh-widgets` and `bokeh-jqwidgets` where the latter has all the jq stuff bundled inside as now 
* building/bundling our own `bokeh-jqui` that we publish can be included when necessary.